### DISCUSSION TITLE
pB issue in output files for 15N CEST

### DISCUSSION THREAD

**Discussion by jbNa95** [gbouvignies/ChemEx#212] [Q&A] [answered]
Hello all, 

I'm trying 15N CEST data analysis using ChemEx in the NMR box to investigate the residue-specific Kex and pB values. However, the pB values acquired from ChemEx were significantly inadequate. The pB values look approximately 48% in the fitting graph. However, the pB value acquired from ChemEx fitting indicates 2.45%. How do I resolve this issue? 

I did run the "run. sh"  (include -e. -p, -o, -d 2st_rs). 
The parameter. toml is here, 

[GLOBAL]
#PB     =  0.41
#KEX_AB = 70.0
TAUC_A = 6.4

[CS_A]
K6N        =  121.848
V7N        =  122.899
I8N        =  124.934
... 
[DW_AB]
K6N        =   -2.984
V7N        =   -2.136
I8N        =   -7.562
...

The experiment. toml is here, 

[experiment]
name         = "cest_15n"
time_t1      = 2
carrier      = 117.000
b1_frq       = 15
b1_inh_scale = inf

[conditions]
h_larmor_frq = 850.51
# p_total     = 2.0e-3
# sample      = "A39G FF domain"
# temperature = 1.0

[data]
path           = "./new_folder/"
error          = "scatter"
#filter_offsets = [[-1000.0, 2090.0]]

  [data.profiles]
K6N = "K6N-H.out"
V7N = "V7N-H.out"
I8N  = "I8N-H.out"
...

I don't know the global Pb and Kex settings, so I set the default.
If I did set the wrong parameter setting value in the two different .toml files, please let me know. 

Best regards, 
Jongbum

![CEST_Pb](https://github.com/gbouvignies/ChemEx/assets/164289922/8276dd73-50c0-4409-9906-7b745b03354f)

**Comment by gbouvignies** [comment]
Hello Jongbum,

The magnitude of the dips is not directly proportional to the state populations. Reviewing the experimental setup, it appears you are applying irradiation for 2 full seconds, which is sufficient to saturate the minor dips.

Moreover, the parameter outputs indicate that all parameters, except for the chemical shift of the minor state, are not well-defined. The errors are comparable to the fitted values, with R2_B converging to 0/s, possibly due to insufficient data to accurately model your system. Incorporating additional B1 field datasets could enhance residue-specific fitting.

Improving the fit might also be achieved by equating R2_B with R2_A, implementable by including the line below in your method file:

```toml
[STEP_NAME]
CONSTRAINTS=["[R2_B]=[R2_A]"]
```

See https://gbouvignies.github.io/ChemEx/docs/user_guide/fitting/method_files#constraints.

**Reply by jbNa95** [reply]
Thank you for your quick response and clear feedback. 

However, I could not understand a few parts of my 15N CEST results obtained through Chemex software.
As you mentioned, the dip intensity does not directly affect the Pb or Kex values. 
Despite the goodness of graph fitting (Reduced-chi-square: 0.7), PB values came out around 99.9% or 100%. I suspect it might be an error.
So, which specific aspects should be primarily examined to determine the confidence of the PB or Kex results?


Your insights on this matter would be greatly appreciated. 
![99% PB ](https://github.com/gbouvignies/ChemEx/assets/164289922/371084b8-bc5e-4ff6-aea0-e85e39b5c37f)

**Comment by gbouvignies** [comment]
If parameters are ill-defined and/or correlated, consider fixing one of them. For instance, setting PB to 0.1 can be explored by modifying the 'parameter.toml' file and adding the line:

```toml
[STEP_NAME]
FIX=["PB"]
```

in the 'method.toml' file.

If the fit quality does not decrease significantly, acquiring more data may be necessary to better constrain the system.

**Reply by jbNa95** [reply]
Thank you very much for your quick response. 

Then, what criteria indicate that PB or Kex values are considered accurate?

**Comment by viochemist** [comment, marked as answer]
Hello,

I thought I'd chime in with my 2 cents. Your fitted 'kex' value is only
2.3/s. Based on the screenshot, your step size seems to be about 0.5ppm so
about 30-50Hz I would guess, with B1 probably about half that. In my
experience, to accurately determine a kex so low you would need much weaker
B1 field values. Fortunately, I do not think that is necessary in this case
because you are most likely in a local minimum. I would recommend 2 things
in fitting and 1 for measurement:

1) Run a grid search over the parameters kex (10-1000) and pB (0.1-10%). I
fully expect you will find a minimum with a more reasonable kex and pb.
This is accomplished with a method step such as:

[grid_step]
FIX = ['R2_B']
FIT = ['R2_A', 'DW_AB']

GRID = [
"[KEX_AB]  = log(10, 1000, 21)",
"[PB]  = log(0.001, 0.1, 21)"
]

2) Run statistics on the final result to see how strongly the parameters
are correlated. This is done with

[stat_step]
STATISTICS = {'BS' = 100, 'MC' = 100, 'BSN' = 100}

BS is for bootstrap statistics, MC is for monte carlo, and BSN is a
residue-based bootstrap (useful if you have many residues that experience
exchange). You can choose one or more of these and then plot the results in
your favorite spreadsheet program or matplotlib.

3) On the experimental side, do run a second B1 field that's ~2-3x stronger
than your current B1 field. If you know your kex after the above steps, you
can make a more rational choice of B1 (see Vallurupalli, Pramodh, et al.
"Probing conformational dynamics in biomolecules via chemical exchange
saturation transfer: a primer." *Journal of biomolecular NMR* 67 (2017):
243-271.)

I hope this helps you on your way. There are a lot of things you can do
with chemex to gain confidence in your results.

P.S. I apologize if my syntax above is not current ... for that I would
need Guillaume to chime in. This all works as of Chemex 2022.0.0. I
definitely need to update myself! :-)

All the best,

Alex Hansen
